mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -227,9 +657,42 @@ impl TableSignature for NftTxHistoryTable {
     fn on_upgrade_needed(upgrader: &DbUpgrader, old_version: u32, new_version: u32) -> OnUpgradeResult<()> {
         if let (0, 1) = (old_version, new_version) {
             let table = upgrader.create_table(Self::table_name())?;
+            table.create_multi_index(
+                Self::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX,
+                &["chain", "token_address", "token_id"],
+                false,
+            )?;
             table.create_multi_index(Self::CHAIN_TX_HASH_INDEX, &["chain", "transaction_hash"], true)?;
+            table.create_multi_index(Self::CHAIN_BLOCK_NUMBER_INDEX, &["chain", "block_number"], false)?;
+            table.create_index("block_number", false)?;
             table.create_index("chain", false)?;
         }
         Ok(())
     }
 }
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct LastScannedBlockTable {
+    chain: String,
+    last_scanned_block: BeBigUint,
+}
+
+impl TableSignature for LastScannedBlockTable {
+    fn table_name() -> &'static str { "last_scanned_block_table" }
+
+    fn on_upgrade_needed(upgrader: &DbUpgrader, old_version: u32, new_version: u32) -> OnUpgradeResult<()> {
+        if let (0, 1) = (old_version, new_version) {
```

Comment:
this check is duplicated all over the codebase, please create helper function for this under `common` module

Reply:
will add it tomorrow, if you dont mind

Reply:
@ozkanonur just a question, do you want me to refactor this check for all code base, or just for the current PR and its related code and finish the rest of the code base in a separate PR? 
Just first variant will increase code changes that should be reviewed in this PR, but if you ok with this I will do it.

Reply:
You can keep it as it is now, even if you make changes. Proper pagination will be implemented later for all the crates/protocols in the near future.